flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -426,6 +426,93 @@ static bool hasPointerType(mlir::Type type) {
   return type.isa<fir::PointerType>();
 }
 
+// This is a NF performance hack. It makes a simple test that the slices of the
+// load, \p ld, and the merge store, \p st, are trivially mutually exclusive.
+static bool mutuallyExclusiveSliceRange(ArrayLoadOp ld, ArrayMergeStoreOp st) {
+  // If the same array_load, then no further testing is warranted.
+  if (ld.getResult() == st.original())
+    return false;
+
+  auto getSliceOp = [](mlir::Value val) -> fir::SliceOp {
+    if (!val)
+      return {};
```

Comment:
I much prefer `return false` to return {}`.

Reply:
This does not return a `bool`.

Reply:
Oops.  Thanks for clarifying.